### ISSUE TITLE
reddark does not properly identify restricted subs

### ISSUE THREAD

**Issue by zsarge** [Tanza3D/reddark#16]
For the reasons outlined by the moderator [u/RamsesThePigeon](https://www.reddit.com/user/RamsesThePigeon/) in [this post](https://www.reddit.com/r/Save3rdPartyApps/comments/141r2mn/a_moreeffective_alternative_to_going_private/), some subreddits are not going completely dark, but instead protesting by going restricted.

For example, [r/edc](https://www.reddit.com/r/edc) [announced](https://www.reddit.com/r/EDC/comments/145f05n/we_are_going_dark_early_in_response_to_the_joke/) they were going restricted in response to the unsatisfying AMA from u/spez. However, reddark still identifies them as "public", despite them joining the protest.

I think it can be debated whether it's worth making a distinction between going private and going restricted, but either way, I don't think the sub should be marked as public.

(Thanks so much for the site btw :sparkles: Such a great idea :smile:)

**Comment by Tanza3D** [comment]
yep, i need to add an alternate "restricted" tag or such to detect, plan to do this tomorrow morning!

**Comment by Tanza3D** [comment]
i've looked into it and sadly can't quite see a way to detect when they're restricted like so, will keep trying

**Comment by adeleke5140** [comment]
@Tanza3D I believe that restricted subs are going private. Maybe that could help?

**Comment by Tanza3D** [comment]
yeah, if they go private they'll show up as privated, but if they just disable submissions and such then i can't see any way to detect that with the data i get, atleast

**Comment by judge-red** [comment]
I'm not a developer, so I can't offer a PR, but if I understand your code correctly, you're fetching `reddit.com/r/${subreddit}.json` and then built some logic to determine whether it's public or private.

IF it's what you call "public", look for `subreddit_type` in the json data and you'll get either `public` or `restricted` (or I think there's other types, there could be all of `gold_restricted`, `archived`, `restricted`, `private`, `employees_only`, `gold_only`, `public`, `user` - see `type` at https://www.reddit.com/dev/api/#POST_api_site_admin).

So this should help you improve "public" detection and add more non-private statuses, I hope.

**Comment by judge-red** [comment]
Are you sure about that? r/games is not restricted, but `restrict_posting` is set to `true` in https://www.reddit.com/r/games/about.json
